### PR TITLE
chore: avoid flaky ntp test on devel ubuntu releases

### PR DESCRIPTION
**Pull request by blackboxsw**
## Proposed Commit Message

```
The ntp package (via ntpsec) occasionally has dependency resolution issues in Ubuntu development releases when new versions of systemd-timesyncd or ntp are released into the archive and cloud images are launched from build serial numbers which do not yet contain the updated systemd-timesyncd package.

Update tests/integration_tests/modules/test_ntp_servers.py on UBUNTU_DEVEL series by providing additional bootcmd to ensure systemd-timesyncd is up to date before any cc_ntp operations take place.

This avoids apt package resolution conflicts when stale ubuntu cloudimages do not yet have knowledge of systemd-timesyncd packages when trying to install ntp/ntpsec.
```

## Additional Context

Intermittent yet repeated ntp error on ubuntu devel releases
https://jenkins.canonical.com/server-team/view/cloud-init/job/cloud-init-integration-oracular-lxd_container/48/
```
tests.integration_tests.modules.test_ntp_servers.TestNtpServers.test_ntp_installed (from pytest)
Failing for the past 3 builds (Since Unstable
[#46](https://jenkins.canonical.com/server-team/view/cloud-init/job/cloud-init-integration-oracular-lxd_container/46/) )
[Took 20 sec.](https://jenkins.canonical.com/server-team/view/cloud-init/job/cloud-init-integration-oracular-lxd_container/48/testReport/junit/tests.integration_tests.modules.test_ntp_servers/TestNtpServers/test_ntp_installed/history)
Error Message

AssertionError: assert False
 +  where False = ''.ok
 +    where '' = execute('ntpd --version')
 +      where execute = <tests.integration_tests.instances.IntegrationInstance object at 0x7fb95540a0b0>.execute
```

## Test Steps
```
CLOUD_INIT_OS_IMAGE=noble CLOUD_INIT_CLOUD_INIT_SOURCE=IN_PLACE tox -e integration-tests -- tests/integration_tests/modules/test_ntp_servers.py::TestNtpServers
CLOUD_INIT_OS_IMAGE=oracular CLOUD_INIT_CLOUD_INIT_SOURCE=IN_PLACE tox -e integration-tests -- tests/integration_tests/modules/test_ntp_servers.py::TestNtpServers
```
## Merge type

- [x] Squash merge using "Proposed Commit Message"
- [ ] Rebase and merge unique commits. Requires commit messages per-commit each referencing the pull request number (#<PR_NUM>)

### CODE REVIEW

File: tests/integration_tests/modules/test_ntp_servers.py [hunk 1]
```diff
@@ -28,31 +30,61 @@
...
+    packaging conflicts as both ntpsec and systemd-timesyncd
+    Provides: ntp
+
+    To avoid this inability to resolve conflicts we need to
+    ensure the installed systemd-timesyncd is updated to the latest
+    published version before trying to install ntpd.
+    Use bootcmd for this work because the cc_ntp module runs before
+    cc_package_update_upgrade_install so we can't add packages:[...]
+    """
+    if CURRENT_RELEASE == UBUNTU_DEVEL:
```
Comment (TheRealFalcon) [inline tests/integration_tests/modules/test_ntp_servers.py:59]: Should we do this regardless of version? I've seen this fail on released noble too.

File: tests/integration_tests/modules/test_ntp_servers.py [hunk 2]
```diff
@@ -28,31 +30,61 @@
       - 172.16.15.15
 """
 
+USER_DATA_UPDATE_TIMESYNCD = """\
+bootcmd:
+- apt-get update -y
```
Comment (TheRealFalcon) [inline tests/integration_tests/modules/test_ntp_servers.py:35]: We may run into apt lock issues here, right?

File: tests/integration_tests/modules/test_ntp_servers.py [hunk 3]
```diff
@@ -28,31 +30,61 @@
...
+    """
+    if CURRENT_RELEASE == UBUNTU_DEVEL:
+        return USER_DATA + USER_DATA_UPDATE_TIMESYNCD
+    return USER_DATA
+
+
 class TestNtpServers:
-    def test_ntp_installed(self, class_client: IntegrationInstance):
+
+    def test_ntp_installed(self, session_cloud: IntegrationCloud, setup_image):
```
Comment (TheRealFalcon) [inline tests/integration_tests/modules/test_ntp_servers.py:66]: `setup_image` shouldn't be needed in these tests.

**Review by TheRealFalcon — APPROVED** [review]
Questions/comments inline. We shouldn't need the `setup_image` fixtures, but it also doesn't hurt things.

**Comment by blackboxsw** [comment]
Closing this PR as further discussion shows that the reason we see this error is due to the ntp transitional package not being cleanly represented for apt resolver whenever systemd-timesyncd and the two packages ntpsec and systemd-timesyncd  both having `Conflicts:/Provides: time-daemon`. Given that ntp is transitional, and ntpsec has been available since debian buster and Ubuntu Focal, we likely want to instead install ntpsec directly instead of the transitional package.

**Pull request closed without merging**